src/Hook.js

```diff
@@ -0,0 +1,182 @@
+import * as Enum from './Enum.js'
+import * as Utility from './Utility.js'
+import * as Context from './Context.js'
+import * as Component from './Component.js'
+import * as Lifecycle from './Lifecycle.js'
+import * as Schedule from './Schedule.js'
+
+/**
+ * @param {any[]} prev
+ * @param {any[]} next
+ * @return {boolean}
+ */
+export function compare (prev, next) {
+	if (prev !== next) {
+		for (var i = 0; i < prev.length; i++) {
+			if (Utility.is(prev[i], next[i]) === false) {
+				return false
+			}
+		}
+	}
+
+	return !!prev
```

Comment:
isn't `!!<Object>` guaranteed to be `true`?

Reply:
This is for !!<undefined> i.e prev === next but prev === undefined in:

```js
useLayout(() => {})
```

When you don't pass dependencies.

Reply:
I see. Seems like this will then throw:
```js
[a, setA] = state(undefined)
b = memo(_=>123, a)
setA(1)
```

Reply:
Why so?

Reply:
Yes. It probably will, that's part of the API contract:

```js
[a, setA] = useState(undefined)
useMemo( _ => 123, [a])
```

The second argument either exists as an array of dependencies or doesn't exist.

```js
useMemo( _ => 123)
```